{"timestamp":"2025-07-13T05:32:39.241470","level":"info","event":"DAG bundles loaded: dags-folder","logger":"airflow.dag_processing.bundles.manager.DagBundlesManager"}
{"timestamp":"2025-07-13T05:32:39.242817","level":"info","event":"Filling up the DagBag from /opt/airflow/dags/data_pipeline_dag.py","logger":"airflow.models.dagbag.DagBag"}
{"timestamp":"2025-07-13T05:32:40.108043Z","level":"error","event":"Traceback (most recent call last):","chan":"stderr","logger":"task"}
{"timestamp":"2025-07-13T05:32:40.108487Z","level":"error","event":"  File \"/opt/airflow/scripts/01_data_visualization.py\", line 5, in <module>","chan":"stderr","logger":"task"}
{"timestamp":"2025-07-13T05:32:40.110239Z","level":"error","event":"    import matplotlib.pyplot as plt","chan":"stderr","logger":"task"}
{"timestamp":"2025-07-13T05:32:40.110517Z","level":"error","event":"ModuleNotFoundError: No module named 'matplotlib'","chan":"stderr","logger":"task"}
{"timestamp":"2025-07-13T05:32:40.184873","level":"error","event":"Task failed with exception","logger":"task","error_detail":[{"exc_type":"CalledProcessError","exc_value":"Command '['python', '/opt/airflow/scripts/01_data_visualization.py']' returned non-zero exit status 1.","exc_notes":[],"syntax_error":null,"is_cause":false,"frames":[{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/sdk/execution_time/task_runner.py","lineno":867,"name":"run"},{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/sdk/execution_time/task_runner.py","lineno":1159,"name":"_execute_task"},{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/sdk/bases/operator.py","lineno":397,"name":"wrapper"},{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/standard/operators/python.py","lineno":216,"name":"execute"},{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/standard/operators/python.py","lineno":239,"name":"execute_callable"},{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/sdk/execution_time/callback_runner.py","lineno":81,"name":"run"},{"filename":"/opt/airflow/dags/data_pipeline_dag.py","lineno":22,"name":"data_visualization"},{"filename":"/usr/local/lib/python3.12/subprocess.py","lineno":571,"name":"run"}],"is_group":false,"exceptions":[]}]}
